packages/twitch-chat-client/src/ChatClient.ts

```diff
@@ -640,9 +645,21 @@ export default class ChatClient extends IRCClient {
 	 */
 	constructor(twitchClient: TwitchClient | undefined, options: ChatClientOptions = {}) {
 		/* eslint-disable no-restricted-syntax */
+		let hostName = null;
+
+		if (options.webSocket === false) {
+			hostName = 'irc.chat.twitch.tv';
+		} else {
+			hostName = 'irc-ws.chat.twitch.tv';
+		}
+
+		if (options.hostName) {
+			hostName = options.hostName;
+		}
+
 		super({
 			connection: {
-				hostName: options.webSocket === false ? 'irc.chat.twitch.tv' : 'irc-ws.chat.twitch.tv',
+				hostName,
```

Comment:
```suggestion
				hostName: options.hostName ?? (options.webSocket === false ? 'irc.chat.twitch.tv' : 'irc-ws.chat.twitch.tv'),
```

This should replace the whole code block before the super call and fix the compile error.

Reply:
Would you really want nullish there? I can see this being pretty common for folx that actually use the feature:

```js
import ChatClient from 'twitch-chat-client'

new ChatClient(twitchClient, {
  hostName: (process.env.NODE_ENV === 'test') ? 'irc.fdgt.dev' : false,
})
```

I think we either want to use an or (`||`) so that falsy values just use the default `hostName`, or we should export the default `hostName` for reuse, e.g.:

```js
import ChatClient, {
  DEFAULT_HOSTNAME,
} from 'twitch-chat-client'

new ChatClient(twitchClient, {
  hostName: (process.env.NODE_ENV === 'test') ? 'irc.fdgt.dev' : DEFAULT_HOSTNAME,
})
```

Thoughts?

Reply:
I've done this with an OR (`||`) for now. Let me know if you'd prefer a different approach.

Reply:
Putting booleans where a string should be doesn't need to be accomodated, it's a user error, easily caught by TypeScript (which wouldn't allow a boolean as per the ChatClientOptions interface) and I don't see many JavaScript people doing this. I would still heavily prefer nullish coalescing.

There is not one default hostname. Exporting both versions separately seems like major overkill for such a feature.